Assets/NuGet/Editor/NugetPackageSource.cs

```diff
@@ -321,27 +322,25 @@ private List<NugetPackage> GetPackagesFromUrl(string url, string password)
 
             try
             {
-                // Mono doesn't have a Certificate Authority, so we have to provide all validation manually.  Currently just accept anything.
-                // See here: http://stackoverflow.com/questions/4926676/mono-webrequest-fails-with-https
-
-                // remove all handlers
-                ServicePointManager.ServerCertificateValidationCallback = null;
-
-                // add anonymous handler
-                ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, policyErrors) => true;
-
                 Stream responseStream = NugetHelper.RequestUrl(url, password, timeOut: 5000);
                 StreamReader streamReader = new StreamReader(responseStream);
 
-                packages = NugetODataResponse.Parse(XDocument.Load(streamReader));                
+                packages = NugetODataResponse.Parse(XDocument.Load(streamReader));
 
                 foreach (var package in packages)
                 {
                     package.PackageSource = this;
                 }
             }
-            catch (System.Exception e)
+            catch (Exception e)
             {
+                var we = e as WebException;
+                if (we != null && we.Status == WebExceptionStatus.TrustFailure && TryHandleTrustFailure())
```

Comment:
With the .NET 3.5 runtime, this occurs with we.Status of SendFailure and we.Message of "Error getting response stream (Write: The authentication or decryption has failed.): SendFailure".

A good way to handle that case may be to have an IsTrustFailure method with a case for detecting the .NET 3.5 error by string in addition to the .NET 4.6 error by Status code.